ml-agents-envs/mlagents/envs/rpc_communicator.py

```diff
@@ -94,7 +95,7 @@ def initialize(self, inputs: UnityInput) -> UnityOutput:
         self.unity_to_external.parent_conn.recv()
         return aca_param
 
-    def exchange(self, inputs: UnityInput) -> UnityOutput:
+    def exchange(self, inputs: UnityInput) -> Optional[UnityOutput]:
```

Comment:
Why this is Optional here?

Reply:
The method can potentially return `None` (line 105), so we have to change the return type. `Optional[UnityOutput]` is shorthand for `Union[None, UnityOutput]`

Reply:
Oh I missed that part of the logic.